packages/yew/src/html/component/scope.rs

```diff
@@ -363,36 +191,309 @@ impl<COMP: BaseComponent> Scope<COMP> {
         &self,
         callback: Callback<T>,
     ) -> Option<(T, ContextHandle<T>)> {
-        self.to_any().context(callback)
+        AnyScope::from(self.clone()).context(callback)
     }
 }
 
 #[cfg(feature = "ssr")]
 mod feat_ssr {
     use super::*;
+    use crate::scheduler;
     use futures::channel::oneshot;
 
+    use crate::html::component::lifecycle::{
+        ComponentRenderState, CreateRunner, DestroyRunner, RenderRunner,
+    };
+
     impl<COMP: BaseComponent> Scope<COMP> {
         pub(crate) async fn render_to_string(self, w: &mut String, props: Rc<COMP::Properties>) {
             let (tx, rx) = oneshot::channel();
-            let initial_render_state = ComponentRenderState::new_ssr(tx);
-
-            self.mount_in_place(initial_render_state, NodeRef::default(), props);
+            let state = ComponentRenderState::Ssr { sender: Some(tx) };
+
+            scheduler::push_component_create(
+                CreateRunner {
+                    initial_render_state: state,
+                    props,
+                    scope: self.clone(),
+                },
+                RenderRunner {
+                    state: self.state.clone(),
+                },
+            );
+            scheduler::start();
 
             let html = rx.await.unwrap();
 
-            let self_any_scope = self.to_any();
+            let self_any_scope = AnyScope::from(self.clone());
             html.render_to_string(w, &self_any_scope).await;
 
             scheduler::push_component_destroy(DestroyRunner {
                 state: self.state.clone(),
+
+                #[cfg(feature = "render")]
                 parent_to_detach: false,
             });
             scheduler::start();
         }
     }
 }
 
+#[cfg(not(any(feature = "ssr", feature = "render")))]
+mod feat_no_render_ssr {
+    use super::*;
+
+    // Skeleton code to provide public methods when no renderer are enabled.
+    impl<COMP: BaseComponent> Scope<COMP> {
+        /// Returns the linked component if available
+        pub fn get_component(&self) -> Option<impl Deref<Target = COMP> + '_> {
+            Option::<&COMP>::None
+        }
+
+        /// Send a message to the component.
+        pub fn send_message<T>(&self, _msg: T)
+        where
+            T: Into<COMP::Message>,
+        {
+        }
+
+        /// Send a batch of messages to the component.
+        ///
+        /// This is slightly more efficient than calling [`send_message`](Self::send_message)
+        /// in a loop.
+        pub fn send_message_batch(&self, _messages: Vec<COMP::Message>) {}
+    }
+}
+
+#[cfg(any(feature = "ssr", feature = "render"))]
+mod feat_render_ssr {
+    use super::*;
+    use crate::scheduler::{self, Shared};
+    use std::cell::Ref;
+
+    #[derive(Debug)]
+    pub(crate) struct MsgQueue<Msg>(Shared<Vec<Msg>>);
```

Comment:
Do we not have MsgQueue on the client side? If I understand this correctly, the MsgQueue is what handles message batching, and thus that will not be available on the client side, correct?

Reply:
I think this is gated under feat_render_ssr?

Which means either render or ssr is enabled, MsgQueue will be enabled.

If no rendering feature is enabled, then it needs to be disabled or clippy will fail as nothing is using it.